packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -288,6 +301,82 @@ export interface AutoScalingGroupProps extends CommonAutoScalingGroupProps {
   readonly role?: iam.IRole;
 }
 
+/**
+ * Group metrics that an Auto Scaling group sends to Amazon CloudWatch.
+ */
+export class GroupMetric {
+
+  /**
+   * All available GroupMetrics.
+   */
+  public static readonly ALL = new Array<GroupMetric>();
+
+  /**
+   * The minimum size of the Auto Scaling group
+   */
+  public static readonly MIN_SIZE = new GroupMetric('GroupMinSize');
+
+  /**
+   * The maximum size of the Auto Scaling group
+   */
+  public static readonly MAX_SIZE = new GroupMetric('GroupMaxSize');
+
+  /**
+   * The number of instances that the Auto Scaling group attempts to maintain
+   */
+  public static readonly DESIRED_CAPACITY = new GroupMetric('GroupDesiredCapacity');
+
+  /**
+   * The number of instances that are running as part of the Auto Scaling group
+   * This metric does not include instances that are pending or terminating
+   */
+  public static readonly IN_SERVICE_INSTANCES = new GroupMetric('GroupInServiceInstances');
+
+  /**
+   * The number of instances that are pending
+   * A pending instance is not yet in service, this metric does not include instances that are in service or terminating
+   */
+  public static readonly PENDING_INSTANCES = new GroupMetric('GroupPendingInstances');
+
+  /**
+   * The number of instances that are in a Standby state
+   * Instances in this state are still running but are not actively in service
+   */
+  public static readonly STANDBY_INSTANCES = new GroupMetric('GroupStandbyInstances');
+
+  /**
+   * The number of instances that are in the process of terminating
+   * This metric does not include instances that are in service or pending
+   */
+  public static readonly TERMINATING_INSTANCES = new GroupMetric('GroupTerminatingInstances');
+
+  /**
+   * The total number of instances in the Auto Scaling group
+   * This metric identifies the number of instances that are in service, pending, and terminating
+   */
+  public static readonly TOTAL_INSTANCES = new GroupMetric('GroupTotalInstances');
+
+  /**
+   * The name of the group metric
+   */
+  public readonly name: string;
+
+  constructor(name: string) {
+    this.name = name;
+  }
```

Comment:
We usually hide this constructor as `private` and open a static method called `custom()` where a user can specify a group metric that is not one of the predefined  ones. We call this pattern is called enum-like class.

Reply:
They still can:

```
asg.emitGroupMetrics(
  new autoscaling.GroupMetric('MyMetric')
);
```
This is the enum like pattern from the `aws-lambda` `runtime` module.

Reply:
I think we may have moved away from using the constructor and moved to using a static method. There should be some other more recent examples. 

The advantage is - if we want to change the set of parameters or create GroupMetric based on some other value, we can just add more static methods, whereas there can be only one constructor.

I'm not totally opposed to this, if you want to still keep the constructor based approach you have here.